shared/p2p/hobbits/rpc_handler.go

```diff
@@ -0,0 +1,296 @@
+	StartSlot uint64 `bson:"start_slot"`
+	Max       uint64 `bson:"max"`
+	Skip      uint64 `bson:"skip"`
+	Direction uint8  `bson:"direction"`
+}
+
+type BlockBodiesResponse struct {
+	Bodies []*v1alpha1.BeaconBlockBody `bson:"bodies"`
+}
+
+type BlockBodyResponse struct {
+	Bodies *v1alpha1.BeaconBlock `bson:"bodies"`
+}
+
+type AttestationRequest struct {
+	Hash []byte `bson:"hash"`
+}
+
+type AttestationResponse struct {
+	Attestation v1alpha1.Attestation `bson:"attestation"`
+}
+
+func (h *HobbitsNode) status(id peer.ID, message HobbitsMessage) error {
+	// does something with the status of the other node
+	responseBody := Status{
+		UserAgent: []byte(fmt.Sprintf("prysm node %s", h.NodeId)),
+		Timestamp: uint64(time.Now().Unix()),
+	}
+
+	body, err := bson.Marshal(responseBody)
+	if err != nil {
+		return errors.Wrap(err, "error marshaling response body")
+	}
+
+	responseMessage := HobbitsMessage{
+		Version:  message.Version,
+		Protocol: message.Protocol,
+		Header:   message.Header,
+		Body:     body,
+	}
+
+	err = h.Server.SendMessage(h.PeerConns[id], encoding.Message(responseMessage))
+	if err != nil {
+		return errors.Wrap(err, "error sending GET_STATUS")
+	}
+
+	return nil
+}
+
+func (h *HobbitsNode) sendHello(id peer.ID, message HobbitsMessage) error {
+	response := h.rpcHello()
+
+	responseBody, err := bson.Marshal(response)
+
+	responseMessage := HobbitsMessage{
+		Version:  message.Version,
+		Protocol: message.Protocol,
+		Header:   message.Header,
+		Body:     responseBody,
+	}
+	log.Trace(responseMessage)
+
+	err = h.Server.SendMessage(h.PeerConns[id], encoding.Message(responseMessage))
+	if err != nil {
+		return errors.Wrap(err, "error sending HELLO")
+	}
+
+	log.Trace("sending HELLO...")
+	return nil
+}
+
+func (h *HobbitsNode) rpcHello() Hello {
+	var response Hello
+
+	response.NodeID = h.NodeId
+	response.BestRoot = h.DB.HeadStateRoot()
+
+	headState, err := h.DB.HeadState(context.Background())
+	if err != nil {
+		log.Printf("error getting HeadState data from db: %s", err.Error())
+	} else {
+		response.BestSlot = headState.Slot // best slot
+	}
+
+	finalizedState, err := h.DB.FinalizedState()
+	if err != nil {
+		log.Printf("error getting FinalizedState data from db: %s", err.Error())
+	} else {
+		response.LatestFinalizedEpoch = finalizedState.Slot / 64 // finalized epoch
+
+		hashedFinalizedState, err := hashutil.HashProto(finalizedState) // finalized root
+		if err != nil {
+			log.Printf("error hashing FinalizedState: %s", err.Error())
+		} else {
+			response.LatestFinalizedRoot = hashedFinalizedState
+		}
+	}
+
+	return response
+}
+
+func (h *HobbitsNode) removePeer(id peer.ID) error {
+	peerToRemove := h.PeerConns[id]
+	delete(h.PeerConns, id)
+
+	err := peerToRemove.Close()
+	if err != nil {
+		return errors.Wrap(err, "error closing connection on peerToRemove")
+	}
+
+	return nil
+}
+
+func (h *HobbitsNode) blockHeadersRequest(id peer.ID, message HobbitsMessage) error { // TODO all block header funcs
```

Comment:
TODO messages must be in the following format `// TODO(1234) todo message` where 1234 is the issue ID tracking the TODO in github.